StripeFinancialConnections/StripeFinancialConnections/Source/Native/AccountPicker/AccountPickerViewController.swift

```diff
@@ -149,47 +150,51 @@ final class AccountPickerViewController: UIViewController {
-                    let shouldSkipAccountSelection = accountsPayload.skipAccountSelection ?? self.dataSource.authSession.skipAccountSelection ?? false
-                    if !accounts.isEmpty {
-                        // the API is expected to never return 0 accounts
-                        self.dataSource
-                            .analyticsClient
-                            .log(
-                                eventName: "polling.accounts.success",
-                                parameters: [
-                                    "duration": Date().timeIntervalSince(pollingStartDate).milliseconds,
-                                    "authSessionId": self.dataSource.authSession.id,
-                                ],
-                                pane: .accountPicker
-                            )
-                    }
                     self.dataSource
                         .analyticsClient
                         .logPaneLoaded(pane: .accountPicker)
 
-                    if accounts.isEmpty {
+                    let accounts = accountsPayload.data
+                    guard !accounts.isEmpty else {
                         // if there were no accounts returned, API should have thrown an error
                         // ...handle it here since API did not throw error
                         self.showAccountLoadErrorView(
                             error: FinancialConnectionsSheetError.unknown(
                                 debugDescription: "API returned an empty list of accounts"
                             )
                         )
-                    } else if shouldSkipAccountSelection {
-                        self.delegate?.accountPickerViewController(
-                            self,
-                            didSelectAccounts: accounts
+                        return
+                    }
+
+                    self.dataSource
+                        .analyticsClient
+                        .log(
+                            eventName: "polling.accounts.success",
+                            parameters: [
+                                "duration": Date().timeIntervalSince(pollingStartDate).milliseconds,
+                                "authSessionId": self.dataSource.authSession.id,
+                            ],
+                            pane: .accountPicker
                         )
-                    } else if self.dataSource.manifest.singleAccount,
+
+                    // note that this uses ?? instead of ||, we do NOT want to skip account selection
+                    // if EITHER of these are true, we only want to skip account selection when
+                    //  `accountsPayload.skipAccountSelection` is true, OR `accountsPayload.skipAccountSelection` nil
+                    // AND `authSession.skipAccountSelection` is true
+                    let skipAccountSelection = (accountsPayload.skipAccountSelection ?? self.dataSource.authSession.skipAccountSelection ?? false)
```

Comment:
no changes requested, but this took me a bit to convince myself this matched the comment 😅 I was just reading the `??`s as C ternary shorthand `?:` but this is good 👍

Reply:
haha yea apologies for this code - I mostly copied this logic + comment from web (part because it makes it easier for me to confirm that it does what it's supposed to do)